ledger/ledgertestutils/ledgertestutils.go

```diff
@@ -47,14 +49,17 @@ func TmpLedger(t testing.TB, lr core.LogicRunner, dir string) (*ledger.Ledger, f
 	pm, err := pulsemanager.NewPulseManager(db)
 	assert.NoError(t, err)
 
+	keeper := nodekeeper.NewNodeKeeper(core.RecordRef{})
+
 	// Init components.
 	mb := testmessagebus.NewTestMessageBus()
-	components := core.Components{MessageBus: mb, LogicRunner: lr}
+	components := core.Components{MessageBus: mb, LogicRunner: lr,
+		ActiveNodeComponent: keeper}
 
 	// Create ledger.
 	l := ledger.NewTestLedger(db, am, pm, jc, handler)
 	err = l.Start(components)
 	assert.NoError(t, err)
 
-	return l, dbcancel
+	return l, dbcancel, keeper
```

Comment:
No need to return keeper here. Pass it via components to this function and use outside.